workflow-trace-viewer/src/jvmMain/kotlin/com/squareup/workflow1/traceviewer/ui/TraceModeToggleSwitch.kt

```diff
@@ -0,0 +1,58 @@
+package com.squareup.workflow1.traceviewer.ui
+
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.padding
+import androidx.compose.material.Switch
+import androidx.compose.material.SwitchDefaults
+import androidx.compose.material.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.runtime.setValue
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.text.font.FontStyle
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.unit.sp
+import com.squareup.workflow1.traceviewer.TraceMode
+
+@Composable
+internal fun TraceModeToggleSwitch(
+  onToggle: () -> Unit,
+  traceMode: TraceMode,
+  modifier: Modifier = Modifier
+) {
+  // File mode is unchecked by default, and live mode is checked.
+  var checked by remember {
+    mutableStateOf(traceMode is TraceMode.Live)
+  }
+
+  Column(
+    modifier = modifier.padding(16.dp),
+    horizontalAlignment = Alignment.CenterHorizontally
+  ) {
+    Switch(
+      checked = checked,
+      onCheckedChange = {
+        checked = it
+        onToggle()
```

Comment:
This is an anti-pattern, since you now have two sources of truth for `traceMode` and violates UDF. You don't need the `checked` property at all, just check the `traceMode` parameter every time. Or do something like
```kotlin
val isLive = traceMode is TraceMode.Live
```
since you check it below as well.